dap-mode.el

```diff
@@ -48,6 +48,69 @@
   :group 'dap-mode
   :type 'boolean)
 
+(defcustom dap-external-terminal '("xterm" "-display" "{title}"
+                                   "-hold" "-e" "sh" "-c" "exec {command}")
+  "Command to launch the external terminal for debugging.
+When specifying that the program be run in an external terminal,
+this command is used to launch it. It shall be a list of strings,
+the first of which is the program to run and the rest of which
+are arguments to pass to it. Special expansion is performed:
+
+{command} will be replaced with the program to execute.
+{display} will be replaced with window title.
+
+See also `dap-default-terminal-kind'."
+  :group 'dap-mode
+  :type '(repeat string))
+
+(defun dap--make-terminal-buffer (title debug-session)
+  "Generate an internal terminal buffer.
+The name is derived from TITLE and DEBUG-SESSION. This function
+should be used in `dap-internal-terminal-*'."
+  (generate-new-buffer
+   (format "*%s %s*"
+           (dap--debug-session-name debug-session)
+           (if title (concat "- " title) "console"))))
+
+(defun dap-internal-terminal-vterm (command title debug-session)
+  (with-current-buffer (dap--make-terminal-buffer title debug-session)
```

Comment:
With the code as it is with fairly recent spacemacs I get:

```
\\{vterm-mode-map}" (interactive) (progn (make-local-variable 'delay-mode-hooks) (let ((delay-mode-hooks t)) (kill-all-local-variables) (setq major-mode 'vterm-mode) (setq mode-name "VTerm") nil (use-local-map vterm-mode-map) (set-syntax-table vterm-mode-syntax-table) (setq local-abbrev-table vterm-mode-abbrev-table) (buffer-disable-undo) (and (boundp 'display-line-numbers) (let ((font-height (expt text-scale-mode-step text-scale-mode-amount))) (setq vterm--linenum-remapping (face-remap-add-relative 'line-number :height font-height)))) (let ((process-environment (append (cons (concat "TERM=" vterm-term-environment-variable) '("INSIDE_EMACS=vterm" "LINES" "COLUMNS")) process-environment)) (process-adaptive-read-buffering nil) (width (max (- (window-body-width) (vterm--get-margin-width)) vterm-min-window-width))) (setq vterm--term (vterm--new (window-body-height) width vterm-max-scrollback vterm-disable-bold-font vterm-disable-underline vterm-disable-inverse-video)) (setq buffer-read-only t) (set (make-local-variable 'scroll-conservatively) 101) (set (make-local-variable 'scroll-margin) 0) (set (make-local-variable 'hscroll-margin) 0) (set (make-local-variable 'hscroll-step) 1) (set (make-local-variable 'truncate-lines) t) (advice--add-function :filter-return (cons #'(lambda nil (advice--buffer-local 'filter-buffer-substring-function)) #'(lambda (gv--val) (advice--set-buffer-local 'filter-buffer-substring-function gv--val))) #'vterm--filter-buffer-substring nil) (setq vterm--process (make-process :name "vterm" :buffer (current-buffer) :command (list "/bin/sh" "-c" (format "stty -nl sane iutf8 erase ^? rows %d columns %d >/dev/null && exec %s" (window-body-height) width vterm-shell)) :coding 'no-conversion :connection-type 'pty :filter #'vterm--filter :sentinel (if (or vterm-exit-functions vterm-kill-buffer-on-exit) (progn #'vterm--sentinel))))) (add-hook 'change-major-mode-hook #'(lambda nil (interactive) (user-error "You cannot change major mode in vterm buffers")) nil t) (vterm--set-pty-name vterm--term (process-tty-name vterm--process)) (process-put vterm--process 'adjust-window-size-function #'vterm--window-adjust-process-window-size) (setq next-error-function 'vterm-next-error-function))) (run-mode-hooks 'vterm-mode-hook)), 1
error in process filter: Wrong number of arguments: ((vterm-mode-abbrev-table vterm-mode-syntax-table t) nil "Major mode for vterm buffer.
```

Reply:
I think it was a blunder related to (vterm-mode *1*). Derived modes don't take an ARG. Fixed (works with Spacemacs, I have the shell layer installed).